### ISSUE TITLE
[v1.2.6-dev] crashes with "std::bad_alloc" when soluble support is configured (distance-to-object=0)

### ISSUE THREAD

**Issue by simonkuehling** [alexrj/Slic3r#2569]
Hi @alexrj,

just a quick crash report for the current 1.2.6-dev @ c9cdae1a9659496f06066eb612e2baf55edfe4c4 - Slic3r exits repeatably when loading the attached simple STL onto the plater after a fresh start (but is working quite nicely with other models...):

```
$ ./slic3r.pl 

(slic3r.pl:20806): Gtk-WARNING **: gtk_widget_size_allocate(): attempt to allocate widget with width -5 and height 24

(slic3r.pl:20806): Gtk-WARNING **: gtk_widget_size_allocate(): attempt to allocate widget with width -5 and height 24

(slic3r.pl:20806): Gtk-WARNING **: gtk_widget_size_allocate(): attempt to allocate widget with width -5 and height 24
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Abgebrochen (Speicherabzug geschrieben)
```

EDIT:
As verified below, it seems to be caused by the setting for soluble support-material (distance-to-object=0) option in my config

Ubuntu 14.04 64bit

[STL file (sphere)](https://www.dropbox.com/s/4l18w9kb3c3wbi5/dual%20extrusion%20-%20sphere%20on%20support%20-%20sphere.stl?dl=1)
[config](https://www.dropbox.com/s/hnuy7g19a6emju9/config.ini?dl=1)

**Comment by justmyopinion** [comment]
When I load your config file i get this warning from Slic3R, may be some limitation to this version:
![cnfg](https://cloud.githubusercontent.com/assets/5621573/5816947/89fadbf4-a0a6-11e4-8c50-a177e3a8e0f1.JPG)

You have selected support and new param: distance to object=0, which seem has some issues and makes slic3R unstable, but I have no more info on that, other than what is reported at this site.
![support](https://cloud.githubusercontent.com/assets/5621573/5816980/d503e47e-a0a6-11e4-931c-305de7165664.JPG)

If this param is different than 0 it seems to slice normally.

**Comment by simonkuehling** [comment]
hi @justmyopinion - thanks for verifying my report!
the config file is actually exported from the selected presets currently used (for easier debugging), but you're right - this does not work that well with mutli-extruder configurations.

Anyway: i can confirm that it is caused by the just recently introduced "soluble" z-distance support option (=0) - and changed the issue title accordingly

**Comment by Vicious-one** [comment]
Seems the same: https://github.com/alexrj/Slic3r/issues/2567

**Comment by alexrj** [comment]
I can only reproduce this by opening the 3D preview tab. If I don't do that, I get no crash even when exporting the G-code. Can you confirm it's the same for you?

**Comment by Vicious-one** [comment]
I'm getting crashes even if I do not open any tab except for 2d plater.

**Comment by simonkuehling** [comment]
Like @Vicious-one this also happens without opening the 3D preview - fresh start, 2d plater, add part, no further interaction (but background processing activated, of course)...

Anything else we can deliver for debugging?

**Comment by alexrj** [comment]
Weird, I got that error with that model and those settings only in the 3D preview rendering code. I fixed it now. I don't understand why you got that error without opening 3D preview...

Can you git pull and rebuild now? I wonder if that's now fixed...

**Comment by Vicious-one** [comment]
I confirm the crash does not occur anymore for me.

**Comment by gege2b** [comment]
hey, it seems to fix #2579 as well !

**Comment by simonkuehling** [comment]
Yes, it is working now! Great.
